src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -159,73 +163,77 @@ func (r *StatefulSetResource) ensureRedpandaGroupsReady(
 	}
 
 	headlessServiceWithPort := fmt.Sprintf("%s:%d", r.serviceFQDN,
-		r.pandaCluster.Spec.Configuration.KafkaAPI.Port)
+		r.pandaCluster.Spec.Configuration.AdminAPI.Port)
 
-	addresses := []string{fmt.Sprintf("%s-%d.%s", sts.Name, ordinal, headlessServiceWithPort)}
+	address := fmt.Sprintf("%s-%d.%s%s", sts.Name, ordinal, headlessServiceWithPort, "/v1/status/ready")
 
-	return r.queryRedpandaForTopicMembers(ctx, addresses, r.logger)
+	return r.queryRedpandaStatus(ctx, address)
 }
 
-// Used as a temporary indicator that Redpanda is ready until a health
-// endpoint is introduced or logic is added here that goes through all topics
-// metadata.
-func (r *StatefulSetResource) queryRedpandaForTopicMembers(
-	ctx context.Context, addresses []string, logger logr.Logger,
+// Temporarily using the status/ready endpoint until we have a specific one for upgrading.
+func (r *StatefulSetResource) queryRedpandaStatus(
+	ctx context.Context, address string,
 ) error {
-	logger.Info("Connect to Redpanda broker", "broker", addresses)
-
-	conf := sarama.NewConfig()
-	conf.Version = sarama.V2_4_0_0
-	conf.ClientID = "operator"
-	conf.Admin.Timeout = time.Second
+	client := &http.Client{Timeout: adminAPITimeout}
+	protocol := "http"
 
 	// TODO right now we support TLS only on one listener so if external
 	// connectivity is enabled, TLS is enabled only on external listener. This
 	// will be fixed by https://github.com/vectorizedio/redpanda/issues/1084
-	if !r.pandaCluster.Spec.ExternalConnectivity.Enabled && r.pandaCluster.Spec.Configuration.TLS.KafkaAPI.Enabled {
+	if !r.pandaCluster.Spec.ExternalConnectivity.Enabled && r.pandaCluster.Spec.Configuration.TLS.AdminAPI.Enabled {
```

Comment:
now that we have admin api with multiple interfaces, should we update this PR too ? 

maybe `choose` the internal IP (but keep the tls check) so that at least the traffic doesn't go out through the internet.

Reply:
We have an external listener but do not expose it in the API yet. This check is based on the API fields. So, we have the following convention: if TLS is enabled we apply it to the external listener. If there is no external listener (external connectivity is disabled), we apply the TLS to the internal listener.

Once #1163 is merged, we'll update this line and be explicit about the listener.

Reply:
But just to be sure, we are using the internal listener here.

Reply:
makes sense.